video/out/hwdec/hwdec_cuda.c

```diff
@@ -57,6 +57,25 @@ int check_cu(const struct ra_hwdec *hw, CUresult err, const char *func)
 
 #define CHECK_CU(x) check_cu(hw, (x), #x)
 
+// Unfortunately, calling cuInit wakes up nvidia dgpus from sleep.
+// That's a blocking operation which can take a few seconds. It also increases power consumption.
+// Avoid this by delaying calling it until absolutely necessary. This lets us bail out from
+// using the hwdec early, without waking up the gpu.
+int cuda_priv_init(const struct ra_hwdec *hw)
+{
+    struct cuda_hw_priv *p = hw->priv;
+    CudaFunctions *cu = p->cu;
+    int ret = 0;
+
+    if (p->initialized)
+        return ret;
+
+    ret = CHECK_CU(cu->cuInit(0));
+    p->initialized = true;
```

Comment:
Should `p->initialized` reflect actual initialization state?

Reply:
Oh yeah, good catch. I shouldn't be setting it as initialized unless it's actually successful.

Reply:
Fixed with b80b40fc19f44f6a1ec82747bcf7c852ae7ed0e6